apps/ui/src/components/BottomAppBar.tsx

```diff
@@ -0,0 +1,466 @@
+import LinkOffIcon from "@mui/icons-material/LinkOff";
+import Paper from "@mui/material/Paper";
+import PlayArrowIcon from "@mui/icons-material/PlayArrow";
+import Popover from "@mui/material/Popover";
+import StopIcon from "@mui/icons-material/Stop";
+import RefreshIcon from "@mui/icons-material/Refresh";
+import CloudUploadIcon from "@mui/icons-material/CloudUpload";
+import List from "@mui/material/List";
+import ListItem from "@mui/material/ListItem";
+
+import { gql, useMutation } from "@apollo/client";
+import { useStore } from "zustand";
+
+import { RepoContext } from "../lib/store";
+import {
+  ListItemIcon,
+  ListItemButton,
+  Stack,
+  ListItemText,
+} from "@mui/material";
+import { myNanoId } from "../lib/utils/utils";
+import { match } from "ts-pattern";
+
+const RuntimeItem = ({ runtimeId }) => {
+  const store = useContext(RepoContext)!;
+  const runtimeMap =
+    useStore(store, (state) => state.getRuntimeMap()) || new Map();
+  // Observe runtime change
+  const runtimeChanged = useStore(store, (state) => state.runtimeChanged);
+  // A dummy useEffect to indicate that runtimeChanged is used.
+  useEffect(() => {}, [runtimeChanged]);
+  const activeRuntime = useStore(store, (state) => state.activeRuntime);
+  const setActiveRuntime = useStore(store, (state) => state.setActiveRuntime);
+  const runtime = runtimeMap.get(runtimeId) || { wsStatus: "unknown" };
+  const repoId = useStore(store, (state) => state.repoId);
+
+  const [requestKernelStatus] = useMutation(
+    gql`
+      mutation RequestKernelStatus($runtimeId: String) {
+        requestKernelStatus(runtimeId: $runtimeId)
+      }
+    `,
+    { context: { clientName: "spawner" } }
+  );
+  const [interruptKernel] = useMutation(
+    gql`
+      mutation InterruptKernel($runtimeId: String) {
+        interruptKernel(runtimeId: $runtimeId)
+      }
+    `,
+    { context: { clientName: "spawner" } }
+  );
+  const [killRuntime] = useMutation(
+    gql`
+      mutation KillRuntime($runtimeId: String, $repoId: String) {
+        killRuntime(runtimeId: $runtimeId, repoId: $repoId)
+      }
+    `,
+    { context: { clientName: "spawner" } }
+  );
+
+  const [disconnectRuntime] = useMutation(
+    gql`
+      mutation DisconnectRuntime($runtimeId: String, $repoId: String) {
+        disconnectRuntime(runtimeId: $runtimeId, repoId: $repoId)
+      }
+    `,
+    { context: { clientName: "spawner" } }
+  );
+
+  const runtimeStatusColors = { idle: "green", busy: "yellow", default: "red" };
+  const runtimeStatus = runtime.status ?? "default";
+  const wsStatusConfig = {
+    connected: { title: "connected", color: "green" },
+    connecting: { title: "connecting", color: "yellow" },
+    default: { title: "disconnected", color: "red" },
+  };
+  const wsStatus = wsStatusConfig[runtime.wsStatus || "default"];
+
+  return (
+    <ListItem key={runtimeId} disablePadding sx={{ margin: 0 }}>
+      {activeRuntime === runtimeId && (
+        <Tooltip title={wsStatus.title}>
```

Comment:
Crash when deleting a runtime.

<img width="635" alt="Screenshot 2023-09-01 at 12 54 55 AM" src="https://github.com/codepod-io/codepod/assets/4576201/c0965e73-bf63-4a9f-956f-f7ad2fe1cb6d">

Reply:
Is it tested after merged PR #501 ? it works fine on my end though.

Reply:
[3/3] And finally, you access `wsStatus.title` which crashes with "reading properties of undefined" error.